language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/CommandSubmissionClientImpl.java

```diff
@@ -30,11 +30,11 @@ public CommandSubmissionClientImpl(
       @NonNull String ledgerId,
       @NonNull Channel channel,
       Optional<String> accessToken,
-      Optional<Deadline> deadline) {
+      ClientInterceptor[] interceptors) {
     this.ledgerId = ledgerId;
     this.serviceStub =
         StubHelper.authenticating(
-            CommandSubmissionServiceGrpc.newFutureStub(channel).withDeadline(deadline.orElse(null)),
+            CommandSubmissionServiceGrpc.newFutureStub(channel).withInterceptors(interceptors),
```

Comment:
I think the best solution is to:
1. revert to accept an optional deadline
2. store the deadline in the object
3. move the `withDeadline` call to when the call stub is created as part of `submit` (should be around line 65)

Reply:
I agree. However since a `Deadline` object represents an absolute deadline (i.e. it starts ticking as soon as it is created), it is not suitable to reuse the same instance in multiple requests. So I used `Duration` and created deadline per request.

Reply:
I closed the PR and created a new one https://github.com/digital-asset/daml/pull/10713